src/test/java/teammates/storage/sqlapi/FeedbackSessionsDbTest.java

```diff
@@ -0,0 +1,255 @@
+        doReturn(feedbackSession).when(feedbackSessionsDb).getFeedbackSession(sessionName, courseId);
+
+        feedbackSessionsDb.restoreDeletedFeedbackSession(sessionName, courseId);
+
+        assertNull(feedbackSession.getDeletedAt());
+        mockHibernateUtil.verify(() -> HibernateUtil.merge(feedbackSession), times(1));
+    }
+
+    @Test
+    public void testRestoreDeletedFeedbackSession_sessionDoesNotExist_throwsEntityDoesNotExistException()
+            throws EntityDoesNotExistException {
+        FeedbackSession feedbackSession = createnewSession();
+        String sessionName = feedbackSession.getName();
+        String courseId = feedbackSession.getCourse().getId();
+        doReturn(null).when(feedbackSessionsDb).getFeedbackSession(sessionName, courseId);
+
+        assertThrows(EntityDoesNotExistException.class,
+                () -> feedbackSessionsDb.restoreDeletedFeedbackSession(sessionName, courseId));
+        mockHibernateUtil.verify(() -> HibernateUtil.merge(feedbackSession), never());
+    }
+
+    @Test
+    public void testSoftDeleteFeedbackSession_success() throws EntityDoesNotExistException {
+        FeedbackSession feedbackSession = createnewSession();
+        String sessionName = feedbackSession.getName();
+        String courseId = feedbackSession.getCourse().getId();
+        doReturn(feedbackSession).when(feedbackSessionsDb).getFeedbackSession(sessionName, courseId);
+
+        feedbackSessionsDb.softDeleteFeedbackSession(sessionName, courseId);
+
+        assertNotNull(feedbackSession.getDeletedAt());
+        mockHibernateUtil.verify(() -> HibernateUtil.merge(feedbackSession), times(1));
+    }
+
+    @Test
+    public void testSoftDeleteFeedbackSession_sessionDoesNotExist_throwsEntityDoesNotExistException()
+            throws EntityDoesNotExistException {
+        FeedbackSession feedbackSession = createnewSession();
+        String sessionName = feedbackSession.getName();
+        String courseId = feedbackSession.getCourse().getId();
+        doReturn(null).when(feedbackSessionsDb).getFeedbackSession(sessionName, courseId);
+
+        assertThrows(EntityDoesNotExistException.class,
+                () -> feedbackSessionsDb.restoreDeletedFeedbackSession(sessionName, courseId));
+        mockHibernateUtil.verify(() -> HibernateUtil.merge(feedbackSession), never());
+    }
+
+    private FeedbackSession createnewSession() {
+        Instant startTime = TimeHelperExtension.getInstantDaysOffsetFromNow(1);
+        Instant endTime = TimeHelperExtension.getInstantDaysOffsetFromNow(7);
+        return new FeedbackSession("name",
+                getTypicalCourse(),
+                "test@email.com",
+                "instructions",
+                startTime,
+                endTime,
+                startTime,
+                endTime,
+                Duration.ofMinutes(5),
+                false,
+                false,
+                false);
+    }
```

Comment:
let's move this to the BaseTestCase, and replace it with getTypicalFeedbackSession (i am aware that the current getTypicalFeedbackSession results in an invalid FeedbackSession, feel free to fully replace the method with this as it should not affect the other test cases currently)